client/app/scripts/superdesk-desks/directives.js

```diff
@@ -189,21 +187,29 @@ define([
                             scope.editStage = null;
                             scope.select(item);
                             scope.message = null;
-                        }, function(response) {
-                            scope.message = gettext('There was a problem, stage not added.');
-                        });
+                        }, errorMessage);
                     } else {
                         api('stages').save(orig, scope.editStage)
                         .then(function(item) {
                             scope.editStage = null;
                             scope.message = null;
                             scope.select(item);
-                        }, function(response) {
-                            scope.message = gettext('There was a problem, stage was not saved.');
-                        });
+                        }, errorMessage);
                     }
                 };
 
+                function errorMessage(response) {
+                    if (response.data && response.data._issues && response.data._issues.name && response.data._issues.name.unique) {
+                        scope.message = gettext(
```

Comment:
this message can't be translated - it would require a defined translation for each stage name..
better set some `scope._error = true;` and in template:
```
<span ng-show="_error" translate>Stage with name "{{ editStage.name }}" already exists.</span>
```
translate will first replace the string with translation and then it will eval `{{ editStage.name }}`